R/est.incidence.by.R

```diff
@@ -101,6 +101,36 @@ est.incidence.by <- function(
     return(to_return)
   }
 
+  if (!all(is.element(strata, pop_data %>% names()))) {
+    # accommodate multiple strata
+    combined_pattern <- paste(strata, collapse = "|")
```

Comment:
This approach seems sub-optimal, for two reasons:

1. we won't be able to tell which columns partially match each specific missing element of `strata`.
2. we didn't remove the elements of `strata` that already exactly match columns in `pop_data`

More generally, complicated regular expressions are a recipe for bugs; if you think you need to use `|` in a regex, please carefully consider alternative approaches. As always, we want to keep the code as simple as possible; for example, in this case, instead of pasting `strata` together, we could instead individually grep each element of `strata` that isn't exactly in `pop_data`, using a `for` loop, or `lapply()`, or `purrr::map()`, etc. 

Then, we can report the possible partial matches for each missing element of `strata`; the markup capabilities of `cli_abort()` enable us to provide this information in an orderly, structured format. Please give all this a try, but if you get stuck, just let me know and I'll take over.

Reply:
I have made changes to this approach to capture full match, partial match and no match.